app/src/split/bluetooth/central.c

```diff
@@ -126,14 +130,25 @@ static uint8_t split_central_discovery_func(struct bt_conn *conn, const struct b
         if (err) {
             LOG_ERR("Discover failed (err %d)", err);
         }
+    } else if (!bt_uuid_cmp(discover_params.uuid,
+                            BT_UUID_DECLARE_128(ZMK_SPLIT_BT_CHAR_RUN_BEHAVIOR_UUID))) {
+        run_behavior_handle = bt_gatt_attr_value_handle(attr);
     } else {
         subscribe_params.notify = split_central_notify_func;
         subscribe_params.value = BT_GATT_CCC_NOTIFY;
         subscribe_params.ccc_handle = attr->handle;
 
         split_central_subscribe(conn);
 
-        return BT_GATT_ITER_STOP;
+        memcpy(&uuid, BT_UUID_DECLARE_128(ZMK_SPLIT_BT_CHAR_RUN_BEHAVIOR_UUID), sizeof(uuid));
```

Comment:
Is it intended that split_central_process_connection and this function share the static uuid variable? It looks like everything here overwrites the UUID passed to bt_gatt_discover while split_central_process_connection does not. If that's not intended, maybe put separate static variables into each function to ensure they can't both access the same variable.

I'm not sure I fully understand what this code is doing yet, but if split_central_process_connection is always supposed to be discovering ZMK_SPLIT_BT_SERVICE_UUID, it looks like it actually is re-discovering the last thing you tried to discover.

Reply:
This code is cribbed from a discovery example from Zephyr. As I read it, they're trying to reduce memory usage by re-using several variables as they discover/iterate the attributes. Let me review these changes again before we move forward, be sure we're still happy with how this is shaping up.

Reply:
@joelspadin Ok, I've cleaned this up a *lot*, and in the process, I believe found and fixed a split bug that would sometimes cause peripherals not to get subscribed to properly if the central is kept on and the peripheral is reset and reconnected to.

The changes depend on the fix in zmkfirmware/zephyr#3 being merged first, then this can be tested.